GTG/core/tasks2.py

```diff
@@ -0,0 +1,476 @@
+# -----------------------------------------------------------------------------
+# Getting Things GNOME! - a personal organizer for the GNOME desktop
+# Copyright (c) The GTG Team
+#
+# This program is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program.  If not, see <http://www.gnu.org/licenses/>.
+# -----------------------------------------------------------------------------
+
+from uuid import uuid4
+import logging
+from typing import List, Callable
+from enum import Enum
+import re
+import datetime
+from operator import attrgetter
+
+from lxml.etree import Element, SubElement, CDATA
+
+from GTG.core.base_store import BaseStore
+from GTG.core.tags2 import Tag2, TagStore
+from GTG.core.dates import Date
+
+log = logging.getLogger(__name__)
+
+
+# ------------------------------------------------------------------------------
+# REGEXES
+# ------------------------------------------------------------------------------
+
+TAG_REGEX = re.compile(r'^\B\@\w+(\-\w+)*\,+')
+SUB_REGEX = re.compile(r'\{\!.+\!\}')
+
+
+# ------------------------------------------------------------------------------
+# TASK STATUS
+# ------------------------------------------------------------------------------
+
+class Status(Enum):
```

Comment:
In that same idea, I would say that, if we're not planning on doing Status comparison (like `status < Status.DISMISSED` or something), it might be more explicit to use some text enum. Like: 
```python3
class Status(Enum)
    ACTIVE = 'ACTIVE'
    DONE = 'DONE'
    DISMISSED = 'DISMISSED'
```
 This way, dumping values or writing them into xml would show more obviously the actual status.

Reply:
I like jaesivsm's idea, it even makes the xml code simpler